xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -1875,6 +1875,12 @@ bool COutput::Init()
   m_pp->PreInit(m_config, &m_diMethods);
   delete m_pp;
 
+  std::list<EINTERLACEMETHOD> deintMethods;
+  deintMethods.assign(m_diMethods.diMethods, m_diMethods.diMethods + m_diMethods.numDiMethods);
+  deintMethods.push_back(VS_INTERLACEMETHOD_AUTO);
```

Comment:
auto is no deinterling mode and is not automatic either. hence crap.
should be handled similar to sw decoder. if mode is not supportd by vaapi, select default.
make sure to add mode none.